### ISSUE TITLE
Switch to promises ?

### ISSUE THREAD

**Issue by jlatrive** [cojs/co-body#29]
Hello, since thunks are deprecated in favor of promises in co, maybe co-body should switch to promises.
I'm willing to work on a pull request if there is any interest. Just tell me.

**Comment by coderhaoxin** [comment]
I'm +1 for `Promise`, but maybe @fengmk2 thinks  `thunk work fine now`.

**Comment by jlatrive** [comment]
I went ahead and created pull request #30.

**Comment by yoshuawuyts** [comment]
This would be [very nice to have](https://github.com/jonathanong/server-style-guide#routes), especially since `async await` doesn't seem to play ball with the current implementation.

e.g. this doesn't work yet:
```js
async function (req, res) {
  const body = await parse.json(req)
}
```
But from my understanding it will work if it was promise based.

**Comment by fengmk2** [comment]
+1

**Comment by jlatrive** [comment]
Included in 4.0 :smile_cat:
